crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1060,6 +1060,28 @@ async fn filter_payment_methods(
         let parse_result = serde_json::from_value::<PaymentMethodsEnabled>(payment_method);
         if let Ok(payment_methods_enabled) = parse_result {
             let payment_method = payment_methods_enabled.payment_method;
+            let allowed_payment_method_types = if let Some(payment_intent) = payment_intent {
+                payment_intent
+                    .metadata
+                    .as_ref()
+                    .and_then(|masked_metadata| {
+                        let metadata = masked_metadata.peek().clone();
+                        let parsed_metadata: Option<api_models::payments::Metadata> =
+                            match serde_json::from_value(metadata) {
+                                Ok(metadata) => Some(metadata),
+                                _ => None,
+                            };
```

Comment:
```suggestion
                            serde_json::from_value(metadata)
                                .map_err(|error| logger::error!(%error, "Failed to deserialize payment intent metadata"))
                                .ok()
```

@shakthi-22 Don't we need to raise errors in case parsing metadata fails?